src/expandableCalendar/WeekCalendar/index.tsx

```diff
@@ -1,221 +1,236 @@
-import Presenter from './presenter';
+import {UpdateSources} from '../commons';
 import constants from '../../commons/constants';
+import {extractCalendarProps} from '../../componentUpdater';
+import CalendarContext from '../Context';
 
-const NUMBER_OF_PAGES = 2; // must be a positive number
-const applyAndroidRtlFix = constants.isAndroid && constants.isRTL;
+const NUMBER_OF_PAGES = 6; // must be a positive number
+const NUM_OF_ITEMS = NUMBER_OF_PAGES * 2 + 1; // NUMBER_OF_PAGES before + NUMBER_OF_PAGES after + current
+const APPLY_ANDROID_FIX = constants.isAndroid && constants.isRTL;
 
 export interface WeekCalendarProps extends CalendarListProps {
   /** whether to have shadow/elevation for the calendar */
   allowShadow?: boolean;
-  context?: any;
-}
-
-interface State {
-  items: string[];
 }
 
 /**
  * @description: Week calendar component
  * @note: Should be wrapped with 'CalendarProvider'
  * @example: https://github.com/wix/react-native-calendars/blob/master/example/src/screens/expandableCalendar.js
  */
-class WeekCalendar extends Component<WeekCalendarProps, State> {
-  static displayName = 'WeekCalendar';
-
-  static propTypes = {
-    ...CalendarList.propTypes,
-    allowShadow: PropTypes.bool
-  };
-
-  static defaultProps = {
-    firstDay: 0,
-    allowShadow: true
-  };
-
-  style = styleConstructor(this.props.theme);
-  presenter = new Presenter();
-  page = NUMBER_OF_PAGES;
-  // On Android+RTL there's an initial scroll that cause issues
-  firstAndroidRTLScrollIgnored = !applyAndroidRtlFix;
-
-  state: State = {
-    items: this.presenter.getDatesArray(this.props)
-  };
-
-  componentDidUpdate(prevProps: WeekCalendarProps) {
-    const {context, firstDay = 0} = this.props;
-    const {shouldComponentUpdate, getDatesArray, scrollToIndex} = this.presenter;
-
-    if (shouldComponentUpdate(this.props.context, prevProps.context)) {
-      if (!sameWeek(context.date, prevProps.context.date, firstDay) || context.numberOfDays) {
-        // Don't update items if the new date is on the same week
-        this.setState({items: getDatesArray(this.props)});
-        scrollToIndex(false);
+const WeekCalendar = (props: WeekCalendarProps) => {
+  const {
+    calendarWidth,
+    hideDayNames,
+    current,
+    theme,
+  } = props;
+  const context = useContext(CalendarContext);
+  const {allowShadow = true, ...calendarListProps} = props;
+  const {style: propsStyle, onDayPress, firstDay = 0, ...others} = extractCalendarProps(calendarListProps);
+  const {date, numberOfDays, updateSource, setDate, timelineLeftInset} = context;
+  const visibleWeek = useRef(date);
+  const style = useRef(styleConstructor(theme));
+  const items = useRef<string[]>(getDatesArray(current ?? date, firstDay, numberOfDays));
+  const [listData, setListData] = useState(items.current);
+  const changedItems = useRef(constants.isRTL);
+  const list = useRef<FlatList>(null);
+  const currentIndex = useRef(NUMBER_OF_PAGES);
+
+  useEffect(() => {
```

Comment:
Do we need to do this on load as well or just on update? Maybe consider changing to `useDidUpdate` hook instead...